src/autofix/solutions/codeReplacer.ts

```diff
@@ -0,0 +1,336 @@
+		}
+
+		if (!nodeInfo?.node) {
+			throw new Error(`Unable to produce solution for message: ${JSON.stringify({fixHints, position, args})}`);
+		}
+
+		// Calculate the replacement position
+		// It cannot be derived from the fixHintsGenerator as it works on the compiled source file.
+		// Just take the length of the old and the original start position
+		let pos, originalLengthNode = nodeInfo.node;
+
+		// For jQuery function deprecations, the node might only cover
+		// the deprecated method, but not the whole expression.
+		// Find the expression and update the position.
+		if (!ts.isCallExpression(nodeInfo.node) &&
+			ts.isPropertyAccessExpression(nodeInfo.node) &&
+			nodeInfo.node.expression && ts.isCallExpression(nodeInfo.node.expression)) {
+			pos = sourceFile.getLineAndCharacterOfPosition(nodeInfo.node.expression.pos);
+		}
+
+		if (ts.isCallExpression(nodeInfo.node.parent) &&
+			nodeInfo.node.parent.expression === nodeInfo.node) {
+			originalLengthNode = nodeInfo.node.parent;
+		}
+
+		const originalLength = originalLengthNode.getEnd() - originalLengthNode.getStart();
+		const line = pos ? pos.line : (position.line - 1);
+		const column = pos ? (pos.character + 1) : (position.column - 1);
+		const start = sourceFile.getPositionOfLineAndCharacter(line, column);
+		const end = start + originalLength;
+
+		cleanupChangeSet(changeSet, start, end);
+
+		changeSet.push({
+			action: ChangeAction.REPLACE,
+			start,
+			end,
+			value,
+		});
+	}
+}
+
+// Avoid replacements on colliding position.
+// TODO: For future we need to find a way to merge the changes
+// Currently it skips replacements of an imported jQuery module variable as it is unnecessary.
+function cleanupChangeSet(changeSet: ChangeSet[], start: number, end: number) {
+	function isReplacementChangeSet(changeSet: ChangeSet): changeSet is ReplaceChange {
+		return changeSet.action === ChangeAction.REPLACE;
+	}
+
+	let i = 0;
+	while (changeSet[i]) {
+		const isReplacement = isReplacementChangeSet(changeSet[i]);
+		if (!isReplacement) {
+			i++;
+			continue;
+		}
+		const replacementSet = changeSet[i] as ReplaceChange;
+
+		if (start <= replacementSet.start && replacementSet.end <= end) {
+			changeSet.splice(i, 1);
+		} else {
+			i++;
+		}
+	}
+}
+
+function patchMessageFixHints(fixHints?: FixHints, apiName?: string) {
+	if (!fixHints || !("exportCodeToBeUsed" in fixHints) ||
+		!fixHints.exportCodeToBeUsed ||
+		!(typeof fixHints.exportCodeToBeUsed === "object")) {
+		return fixHints;
+	}
+
+	if (apiName === "jQuery.sap.getUriParameters" && fixHints.exportCodeToBeUsed.args?.[0]) {
+		let probingUrl = "";
+		try {
+			new URL(fixHints.exportCodeToBeUsed.args[0].value);
+		} catch (_e) {
+			// Adding a domain to the URL to prevent errors and parse correctly the query string
+			probingUrl = ", \"http://example.com\"";
+		}
+		fixHints.exportCodeToBeUsed.name = `new URL($1${probingUrl}).searchParams`;
```

Comment:
dead code as the replacement is commented out

Reply:
Done